src/app/data.service.ts

```diff
@@ -1,19 +1,25 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { User } from './models/user';
+import { Post } from './models/post';
 
 @Injectable({
   providedIn: 'root'
 })
 export class DataService {
 
+  private rootUrl = 'https://jsonplaceholder.typicode.com';
+
   constructor(private http: HttpClient) {
   }
 
-  getUsersData() {
-    return this.http.get('http://jsonplaceholder.typicode.com/users');
+  getUserData(userId: number = null): Observable<User> {
+    return this.http.get<User>(`${this.rootUrl}/users/${userId != null ? userId : ''}`);
   }
 
-  getPostsData() {
-    return this.http.get('http://jsonplaceholder.typicode.com/posts');
+  getPostData(postId: number = null): Observable<Post> {
```

Comment:
We also added types to our functions so we know what we are getting back.  Technically the return type should be a TypeScript Union type since we don't know if it is a single post or an array.
```Observable<Post | Post[]>```